docs/source/developers/guide/introduction.rst

```diff
@@ -0,0 +1,135 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _introduction:
+
+**********************
+New Contributor's Guide
```

Comment:
I would call it simply "Contributor's guide", because it should be useful not only for _new_ contributors.

Reply:
@amol  what do you think?

Reply:
I think it's reasonable to name it "Contributor's Guide". If we think there is anything too specific to new contributors we can for example have a "Getting Started" section that people can skip

Reply:
Great, I'll rename to "Contributor's guide". Also I'll keep this in mind when structuring the text.

Reply:
> Thanks for starting this!
> 
> I think it is desirable to reframe this a bit before you write too much text:
> 
>     * this "contributor's guide" (or "contribution guide"? @thisisnic  what do you think?) should IMHO focus on the language-independent aspects, whether technical, social or communicational (how to get set up with Git, fork the repo, use JIRA, ask for help, etc.)
> 
>     * all language-specific instructions about building, testing, installing dependencies, etc. should go into language-specific docs (if they are not already there); don't hesitate to open individual JIRAs if such documentation is currently missing!
> 
> 
> Opinions @amol- @jorisvandenbossche ?

I don't think there's enough difference between "contribution guide" and "contributor's guide" for one to be particularly better than the other.

I also agree that the language-specific elements should go into their corresponding docs.

Just a thought - I wonder if some of the original idea behind this initiative is being diluted a little here?  My understanding was that originally this was a guide for new contributors, rather than a new version of the contribution docs, with the point being that there are actually loads of Arrow users who are totally capable of contributing, but maybe don't see themselves as such.  And so, it calls for a gentle introduction, helped by the informal language etc, and extra context/descriptive stuff.

Whereas there is a secondary task of generally improving the contribution docs, in terms of writing more general content, which may be more concise, more formal, and serve the purpose of just providing information.

Things may have evolved from this - feel free to disregard this - but just wanted to check as I feel like there are multiple different approaches here.

Reply:
I do have to agree with @thisisnic on all said. For me it would make sense to

- focus on new contributors, 
- use "gentle" informal language,
- adding extra description to the whole process of contributing
- and linking to existing docs all language-specific elements.

I do not think it will discourage other contributors to look for ideas even if it is targeted at new contributors with not so formal language.

Reply:
I personally agree with @thisisnic and @AlenkaF as well. I think the points in @AlenkaF's comment just above is a nice summary of the goals and scope of the this new contributors guide. 

(and I also agree we should further improve and restructure the general contributors docs that we will be linking to from here)

We should pay attention to not create too much duplication, but having a page that explains a bit more the "why" (the broader context) and not just the mechanical "how" (the steps to build pyarrow) is useful I think.

@AlenkaF it might be useful to document the goals / scope of the new subsection you are adding somewhere (eg in a comment in the file, restructuredtext has syntax to add a comment which is then not rendered in the online version)

Reply:
Generally I agree about reframing this more precisely. Perhaps it would also help to discuss who the typical Arrow contributors are/would be, so that we don't miss our target.

Reply:
I think the main page for which we need to look out in not duplicating too much is the current "Contributing to Apache Arrow" first page of the developer section (https://arrow.apache.org/docs/developers/contributing.html), since this also has eg a section on communication.

For the name of the guide, it could also be something like "Getting started with contributing to Arrow" (or a shorter version)